charts/tidb-cluster/templates/scripts/_start_pd.sh.tpl

```diff
@@ -43,56 +42,36 @@ while true; do
         exit 1
     fi
 
-    source ${ANNOTATIONS} 2>/dev/null
-    if nslookup ${PEER_SERVICE_DOMAIN} 2>/dev/null
+    if nslookup ${domain} 2>/dev/null
     then
-        echo "nslookup domain ${HOSTNAME}.${PEER_SERVICE_NAME}.${NAMESPACE}.svc success"
-
-        if [[ ${ORDINAL} -eq 0 ]]
-        then
-            [[ -z ${bootstrapping:-} ]] && continue
-            [[ ${bootstrapping} == "true" ]] && break
-        fi
-
-        [[ -d /var/lib/pd/member/wal ]] && break
-        wget -qO- ${SERVICE_DOMAIN}:2379/pd/api/v1/members 2>/dev/null
-        [[ $? -eq 0 ]] && break
-        echo "pd cluster is not ready now: ${SERVICE_DOMAIN}"
+        echo "nslookup domain ${domain}.svc success"
+        break
     else
-        echo "nslookup domain ${PEER_SERVICE_DOMAIN} failed" >&2
+        echo "nslookup domain ${domain} failed" >&2
     fi
 done
 
 ARGS="--data-dir=/var/lib/pd \
 --name=${HOSTNAME} \
 --peer-urls=http://0.0.0.0:2380 \
---advertise-peer-urls=http://${HOSTNAME}.${PEER_SERVICE_NAME}.${NAMESPACE}.svc:2380 \
+--advertise-peer-urls=http://${domain}:2380 \
 --client-urls=http://0.0.0.0:2379 \
---advertise-client-urls=http://${HOSTNAME}.${PEER_SERVICE_NAME}.${NAMESPACE}.svc:2379 \
+--advertise-client-urls=http://${domain}:2379 \
 --config=/etc/pd/pd.toml \
 "
 
-replicas=${replicas:-3}
-if [[ ${ORDINAL} -eq 0 && ${bootstrapping:-} == "true" ]]
+if [[ ! -d /var/lib/pd/member/wal ]]
 then
-    ARGS="${ARGS}--initial-cluster=${HOSTNAME}=http://${HOSTNAME}.${PEER_SERVICE_NAME}.${NAMESPACE}.svc:2380"
-else
-    if [[ ${ORDINAL} -eq 0 ]]
-    then
-      TOP=$((replicas-1))
-    else
-      TOP=$((ORDINAL-1))
-    fi
-
-    ARGS="${ARGS}--join="
-    for i in $(seq 0 ${TOP});
-    do
-        [[ ${i} -eq ${ORDINAL} ]] && continue
-        ARGS="${ARGS}http://${SET_NAME}-${i}.${PEER_SERVICE_NAME}.${NAMESPACE}.svc:2380"
-        if [[ ${i} -lt ${TOP} ]]
+    while true; do
```

Comment:
You can use `until` instead of `while`:
```shell
until result=$(wget -qO- -T 3 http://${discovery_url}/new/${domain}:2380 2>/dev/null); do
    echo "waiting for discovery service returns start args ..."
    sleep 2
done
ARGS="${ARGS}${result}"
```